diff --git a/mozjs-sys/mozjs/js/src/jsfriendapi.h b/mozjs-sys/mozjs/js/src/jsfriendapi.h
index 74ceb75f6..cd1bbe6da 100644
--- a/mozjs-sys/mozjs/js/src/jsfriendapi.h
+++ b/mozjs-sys/mozjs/js/src/jsfriendapi.h
@@ -245,6 +245,8 @@ extern JS_PUBLIC_API bool EnqueueJob(JSContext* cx, JS::HandleObject job);
  */
 extern JS_PUBLIC_API void StopDrainingJobQueue(JSContext* cx);
 
+extern JS_PUBLIC_API bool HasJobsPending(JSContext* cx);
+
 extern JS_PUBLIC_API void RunJobs(JSContext* cx);
 
 extern JS_PUBLIC_API JS::Zone* GetRealmZone(JS::Realm* realm);
diff --git a/mozjs-sys/mozjs/js/src/vm/JSContext.cpp b/mozjs-sys/mozjs/js/src/vm/JSContext.cpp
index 3f21e6559..710461bf2 100644
--- a/mozjs-sys/mozjs/js/src/vm/JSContext.cpp
+++ b/mozjs-sys/mozjs/js/src/vm/JSContext.cpp
@@ -816,6 +816,11 @@ JS_PUBLIC_API void js::StopDrainingJobQueue(JSContext* cx) {
   cx->internalJobQueue->interrupt();
 }
 
+JS_PUBLIC_API bool js::HasJobsPending(JSContext* cx) {
+  MOZ_ASSERT(cx->jobQueue);
+  return !cx->jobQueue->empty();
+}
+
 JS_PUBLIC_API void js::RunJobs(JSContext* cx) {
   MOZ_ASSERT(cx->jobQueue);
   cx->jobQueue->runJobs(cx);
