commit cf0eb4cca26acd2eb56c83198382556648c51b03
Author: Christoph Herzog <chris@theduke.at>
Date:   Fri Sep 22 10:43:43 2023 +0200

    fix: Initialize internal spidermonkey job queue

diff --git a/mozjs/src/rust.rs b/mozjs/src/rust.rs
index 92ff809a8..bf172a7ed 100644
--- a/mozjs/src/rust.rs
+++ b/mozjs/src/rust.rs
@@ -337,6 +337,10 @@ impl Runtime {
         let js_context = JS_NewContext(default_heapsize + (ChunkSize as u32), parent_runtime);
         assert!(!js_context.is_null());
 
+        if !crate::jsapi::UseInternalJobQueues(js_context) {
+            panic!("could not initialize internal job queues");
+        }
+
         // Unconstrain the runtime's threshold on nominal heap size, to avoid
         // triggering GC too often if operating continuously near an arbitrary
         // finite threshold. This leaves the maximum-JS_malloc-bytes threshold
