From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <nox@nox.paris>
Date: Mon, 28 Oct 2019 23:12:42 +0100
Subject: [PATCH] Don't use internal encoding_c_mem header


diff --git a/mfbt/moz.build b/mfbt/moz.build
index 62a8c0d991a9..1342c4322b98 100644
--- a/mfbt/moz.build
+++ b/mfbt/moz.build
@@ -9,10 +9,6 @@ with Files("**"):
 
 Library("mfbt")
 
-EXPORTS += [
-    "../third_party/rust/encoding_c_mem/include/encoding_rs_mem.h",
-]
-
 EXPORTS.mozilla = [
     "Algorithm.h",
     "Alignment.h",
