diff --git a/mozglue/misc/Mutex_posix.cpp b/mozglue/misc/Mutex_posix.cpp
index 7378a544f2..2614a2ef3c 100644
--- a/mozglue/misc/Mutex_posix.cpp
+++ b/mozglue/misc/Mutex_posix.cpp
@@ -66,11 +66,9 @@ mozilla::detail::MutexImpl::MutexImpl() {
                     "mozilla::detail::MutexImpl::MutexImpl: "
                     "pthread_mutexattr_settype failed");
 #  elif defined(POLICY_KIND)
-  if (__builtin_available(macOS 10.14, *)) {
-    TRY_CALL_PTHREADS(pthread_mutexattr_setpolicy_np(&attr, POLICY_KIND),
-                      "mozilla::detail::MutexImpl::MutexImpl: "
-                      "pthread_mutexattr_setpolicy_np failed");
-  }
+  (pthread_mutexattr_setpolicy_np(&attr, POLICY_KIND),
+                    "mozilla::detail::MutexImpl::MutexImpl: "
+                    "pthread_mutexattr_setpolicy_np failed");
 #  endif
   attrp = &attr;
 #endif
