From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <nox@nox.paris>
Date: Tue, 29 Oct 2019 10:56:36 +0100
Subject: [PATCH] Work around rust-bindgen#380 for debug builds


diff --git a/mfbt/GuardObjects.h b/mfbt/GuardObjects.h
index 48015f53c532..7592364e332e 100644
--- a/mfbt/GuardObjects.h
+++ b/mfbt/GuardObjects.h
@@ -96,7 +96,7 @@ class GuardObjectNotifier {
   }
 };
 
-class GuardObjectNotificationReceiver {
+class alignas(8) GuardObjectNotificationReceiver {
  private:
   bool mStatementDone;
 
