From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <nox@nox.paris>
Date: Tue, 7 Jan 2020 12:25:35 +0100
Subject: [PATCH] Allow setting _LIBCPP_INLINE_VISIBILITY

https://github.com/servo/mozjs/pull/180

diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 2c09b4f3b0ae..092af7f958a7 100755
--- a/build/moz.configure/toolchain.configure
+++ b/build/moz.configure/toolchain.configure
@@ -1376,8 +1376,14 @@ set_config('COLOR_CFLAGS', color_cflags)
 # this setup.
 
 
-@depends(c_compiler, target)
-def libcxx_override_visibility(c_compiler, target):
+option(env='_LIBCPP_INLINE_VISIBILITY',
+       nargs=1,
+       help='Visibility of libc++ inlines')
+
+@depends('_LIBCPP_INLINE_VISIBILITY', c_compiler, target)
+def libcxx_override_visibility(value, c_compiler, target):
+    if len(value):
+        return value[0]
     if c_compiler.type == 'clang' and target.os == 'Android':
         return namespace(
             empty='',
