From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <nox@nox.paris>
Date: Tue, 7 Jan 2020 12:30:06 +0100
Subject: [PATCH] Always use mozglue


diff --git a/js/src/build/moz.build b/js/src/build/moz.build
index dc58a97e135c..111dbde8c531 100644
--- a/js/src/build/moz.build
+++ b/js/src/build/moz.build
@@ -34,6 +34,7 @@ if CONFIG["JS_SHARED_LIBRARY"]:
         SYMBOLS_FILE = "!symverscript"
 else:
     Library("js")
+    USE_LIBS += ["mozglue"]

 FORCE_STATIC_LIB = True
 STATIC_LIBRARY_NAME = "js_static"
