diff --git a/build/moz.configure/android-ndk.configure b/build/moz.configure/android-ndk.configure
index b6d198b149..9982b0a615 100644
--- a/build/moz.configure/android-ndk.configure
+++ b/build/moz.configure/android-ndk.configure
@@ -21,6 +21,10 @@ option(
     help="location where the Android NDK can be found{|}",
 )
 
+option("--with-android-ndk-version", nargs=1, help="android NDK version")
+
+option("--with-android-platform", nargs=1, help="location of the Android platform")
+
 option("--with-android-toolchain", nargs=1, help="location of the Android toolchain")
 
 option(
@@ -88,17 +92,20 @@ set_config("ANDROID_NDK", ndk)
 add_old_configure_assignment("android_ndk", ndk)
 
 
-@depends(ndk)
+@depends("--with-android-ndk-version", ndk)
 @checking("for android ndk version")
 @imports(_from="__builtin__", _import="open")
 @imports(_from="mozboot.android", _import="NDK_VERSION")
 @imports(_from="mozboot.android", _import="get_ndk_version")
 @imports(_from="mozboot.android", _import="GetNdkVersionError")
-def ndk_version(ndk):
+def ndk_version(value, ndk):
     if not ndk:
         # Building 'js/src' for non-Android.
         return
 
+    if value:
+        return value[0]
+
     try:
         major, minor, human = get_ndk_version(ndk)
     except GetNdkVersionError as e:
@@ -120,13 +127,16 @@ set_config("ANDROID_NDK_MAJOR_VERSION", ndk_version.major)
 set_config("ANDROID_NDK_MINOR_VERSION", ndk_version.minor)
 
 
-@depends(target, android_version, ndk)
+@depends("--with-android-platform", target, android_version, ndk)
 @checking("for android platform directory")
 @imports(_from="os.path", _import="isdir")
-def android_platform(target, android_version, ndk):
+def android_platform(value, target, android_version, ndk):
     if target.os != "Android":
         return
 
+    if value:
+        return value[0]
+
     if "aarch64" == target.cpu:
         target_dir_name = "arm64"
     else:
@@ -267,7 +277,7 @@ option(
 @imports(_from="os.path", _import="isdir")
 def stlport_cppflags(value, ndk):
     if value and len(value):
-        return value.split()
+        return value[0].split()
     if not ndk:
         return
 
