diff --git a/js/moz.configure b/js/moz.configure
index e65adafb73..48ec64d925 100644
--- a/js/moz.configure
+++ b/js/moz.configure
@@ -290,14 +290,14 @@ with only_when("--enable-compile-environment"):
     imply_option("--enable-frame-pointers", True, when=profiling)
 
 
-@depends(profiling, target)
-def imply_vtune(value, target):
+@depends(profiling, target, js_enable_uwp)
+def imply_vtune(value, target, uwp):
     ok_cpu = target.cpu in ["x86", "x86_64"]
     ok_kernel = target.kernel == "WINNT" or (
         target.kernel == "Linux" and target.os == "GNU"
     )
 
-    if value and ok_cpu and ok_kernel:
+    if value and ok_cpu and ok_kernel and not uwp:
         return True
 
 
diff --git a/js/src/builtin/Profilers.cpp b/js/src/builtin/Profilers.cpp
index 53da01dba9..6b2f848510 100644
--- a/js/src/builtin/Profilers.cpp
+++ b/js/src/builtin/Profilers.cpp
@@ -24,8 +24,13 @@
 #endif
 
 #ifdef XP_WIN
-#  include <process.h>
-#  define getpid _getpid
+#  ifdef JS_ENABLE_UWP
+#    include <processthreadsapi.h>
+#    define getpid GetCurrentProcessId
+#  else
+#    include <process.h>
+#    define getpid _getpid
+#  endif
 #endif
 
 #include "js/CharacterEncoding.h"
