From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <nox@nox.paris>
Date: Mon, 13 Jan 2020 15:41:43 +0100
Subject: [PATCH] Ignore dllservices


diff --git a/mozglue/moz.build b/mozglue/moz.build
index e990195da..2361a13e4 100644
--- a/mozglue/moz.build
+++ b/mozglue/moz.build
@@ -13,9 +13,6 @@ if CONFIG["MOZ_LINKER"] or CONFIG["MOZ_WIDGET_TOOLKIT"] == "android":
 if CONFIG["MOZ_WIDGET_TOOLKIT"] == "android":
     DIRS += ["android"]
 
-if CONFIG["OS_TARGET"] == "WINNT":
-    DIRS += ["dllservices"]
-
 DIRS += [
     "baseprofiler",
     "build",
