diff --git a/js/src/build/moz.build b/js/src/build/moz.build
index 3f5dce5f73..9e149581ac 100644
--- a/js/src/build/moz.build
+++ b/js/src/build/moz.build
@@ -91,7 +91,6 @@ GeneratedFile(
     script="/config/run_spidermonkey_checks.py",
     inputs=[
         "!%sjs_static.%s" % (CONFIG["LIB_PREFIX"], CONFIG["LIB_SUFFIX"]),
-        "/config/check_spidermonkey_style.py",
         "/config/check_macroassembler_style.py",
         "/config/check_js_opcode.py",
     ],
