From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <nox@nox.paris>
Date: Tue, 25 Feb 2020 13:36:32 +0100
Subject: [PATCH] Fix MOZ_DBG for MSVC


diff --git a/mfbt/DbgMacro.h b/mfbt/DbgMacro.h
index 9c955c3ee891..d479ffee18d4 100644
--- a/mfbt/DbgMacro.h
+++ b/mfbt/DbgMacro.h
@@ -172,7 +172,7 @@ std::ostream& operator<<(std::ostream& aOut, const T (&aArray)[N]) {
 //                         // MOZ_DBG as an rvalue reference and so wants to
 //                         // invoke B's move constructor to initialize b2
 #ifndef MOZILLA_OFFICIAL
-#  define MOZ_DBG(expression_...) \
+#  define MOZ_DBG(expression_, ...) \
     mozilla::detail::MozDbg(__FILE__, __LINE__, #expression_, expression_)
 #endif
 
@@ -194,7 +194,7 @@ std::ostream& operator<<(std::ostream& aOut, const T (&aArray)[N]) {
 //
 // generates an operator<< that outputs strings like
 // "Point { x = 1.0, y = 2.0 }".
-#define MOZ_DEFINE_DBG(type_, members_...)                                   \
+#define MOZ_DEFINE_DBG(type_, members_, ...)                                   \
   friend std::ostream& operator<<(std::ostream& aOut, const type_& aValue) { \
     return aOut << #type_                                                    \
                 << (MOZ_ARG_COUNT(members_) == 0 ? "" : " { ")               \
