diff --git a/mozjs/js/src/jit/arm64/CodeGenerator-arm64.cpp b/mozjs/js/src/jit/arm64/CodeGenerator-arm64.cpp
index aeb0c97b8..37c987c8d 100644
--- a/mozjs/js/src/jit/arm64/CodeGenerator-arm64.cpp
+++ b/mozjs/js/src/jit/arm64/CodeGenerator-arm64.cpp
@@ -340,8 +340,11 @@ void CodeGenerator::visitMulI(LMulI* ins) {
         Label bailout;
         Label* onOverflow = mul->canOverflow() ? &bailout : nullptr;
 
-        masm.move32(Imm32(constant), destreg);
-        masm.mul32(lhsreg, destreg, destreg, onOverflow);
+        vixl::UseScratchRegisterScope temps(&masm.asVIXL());
+        const Register scratch = temps.AcquireW().asUnsized();
+
+        masm.move32(Imm32(constant), scratch);
+        masm.mul32(lhsreg, scratch, destreg, onOverflow);
 
         if (onOverflow) {
           bailoutFrom(&bailout, ins->snapshot());
