diff --git a/build/moz.configure/toolchain.configure b/build/moz.configure/toolchain.configure
index 814a1655d9..c2bf4cd60b 100755
--- a/mozjs/build/moz.configure/toolchain.configure
+++ b/mozjs/build/moz.configure/toolchain.configure
@@ -137,12 +137,6 @@ with only_when(host_is_osx | target_is_osx):
                 "SDK installation docs: %s"
                 % (version, sdk_min_version, sdk_installation_docs_url)
             )
-        if version > sdk_max_version:
-            die(
-                'SDK version "%s" is unsupported. Please downgrade to version '
-                "%s. See the manual SDK installation docs: %s"
-                % (version, sdk_max_version, sdk_installation_docs_url)
-            )
         return sdk
 
     set_config("MACOS_SDK_DIR", macos_sdk)
