From 2150799db8b3568d03173b90d1b3ba40cf3188c8 Mon Sep 17 00:00:00 2001
From: Ms2ger <Ms2ger@gmail.com>
Date: Wed, 20 Jul 2016 16:55:05 +0200
Subject: Add missing MOZ_ANDROID_CPU_ARCH call in SpiderMonkey
 old-configure.in.


diff --git a/mozjs/js/src/old-configure.in b/mozjs/js/src/old-configure.in
index 30a6a26..02e0346 100644
--- a/mozjs/js/src/old-configure.in
+++ b/mozjs/js/src/old-configure.in
@@ -584,6 +584,7 @@ dnl Android libstdc++, placed here so it can use MOZ_ARCH
 dnl computed above.
 dnl ========================================================
 
+MOZ_ANDROID_CPU_ARCH
 MOZ_ANDROID_STLPORT
 
 dnl ========================================================
-- 
1.9.1

