From f7b74b38405f0d6da47c1b4ecf57390dd0cadf8a Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Fri, 22 Apr 2016 14:07:12 +0200
Subject: Add mozglue to JS_STANDALONE


diff --git a/mozjs/js/src/moz.build b/mozjs/js/src/moz.build
index dab7c8b..f069cd0 100644
--- a/mozjs/js/src/moz.build
+++ b/mozjs/js/src/moz.build
@@ -661,6 +661,9 @@ USE_LIBS += [
     'zlib',
 ]
 
+if CONFIG['JS_STANDALONE']:
+    USE_LIBS += ['mozglue']
+
 if CONFIG['NIGHTLY_BUILD']:
     DEFINES['ENABLE_BINARYDATA'] = True
 
-- 
1.9.1

