diff --git a/mozjs/mozglue/build/BionicGlue.cpp b/mozjs/mozglue/build/BionicGlue.cpp
index b9b882166..710d01f2e 100644
--- a/mozjs/mozglue/build/BionicGlue.cpp
+++ b/mozjs/mozglue/build/BionicGlue.cpp
@@ -17,7 +17,7 @@
 
 #define NS_EXPORT __attribute__((visibility("default")))
 
-#if ANDROID_VERSION < 17 || defined(MOZ_WIDGET_ANDROID)
+#if __ANDROID_API__ < 17 || defined(MOZ_WIDGET_ANDROID)
 /* Android doesn't have pthread_atfork(), so we need to use our own. */
 struct AtForkFuncs {
   void (*prepare)(void);
@@ -64,6 +64,6 @@ private:
 static std::vector<AtForkFuncs, SpecialAllocator<AtForkFuncs> > atfork;
 #endif
 
-#if ANDROID_VERSION < 17 || defined(MOZ_WIDGET_ANDROID)
+#if __ANDROID_API__ < 17 || defined(MOZ_WIDGET_ANDROID)
 extern "C" NS_EXPORT int pthread_atfork(void (*prepare)(void),
                                         void (*parent)(void),
