From d3730ce9debfe919d554b3a32731bfa07f9b1c58 Mon Sep 17 00:00:00 2001
From: Vladimir Vukicevic <vladimir@pobox.com>
Date: Fri, 8 Jan 2016 10:59:15 -0500
Subject: Build mozglue as a static library when building JS_STANDALONE


diff --git a/mozjs/mozglue/build/moz.build b/mozjs/mozglue/build/moz.build
index 1c6f660..64aa31b 100644
--- a/mozjs/mozglue/build/moz.build
+++ b/mozjs/mozglue/build/moz.build
@@ -6,7 +6,8 @@
 
 # Build mozglue as a shared lib on Windows, OSX and Android.
 # If this is ever changed, update MOZ_SHARED_MOZGLUE in browser/installer/Makefile.in
-if CONFIG['OS_TARGET'] in ('WINNT', 'Darwin', 'Android'):
+# If JS_STANDALONE, always build as static.
+if CONFIG['OS_TARGET'] in ('WINNT', 'Darwin', 'Android') and not CONFIG['JS_STANDALONE']:
     SharedLibrary('mozglue')
 else:
     Library('mozglue')
@@ -24,15 +25,15 @@ if CONFIG['MOZ_ASAN']:
         'AsanOptions.cpp',
     ]
 
-if CONFIG['OS_TARGET'] == 'WINNT':
-    DEFFILE = 'mozglue.def'
-    # We'll break the DLL blocklist if we immediately load user32.dll
-    DELAYLOAD_DLLS += [
-        'user32.dll',
-    ]
-
 if not CONFIG['JS_STANDALONE']:
 
+    if CONFIG['OS_TARGET'] == 'WINNT':
+        DEFFILE = 'mozglue.def'
+        # We'll break the DLL blocklist if we immediately load user32.dll
+        DELAYLOAD_DLLS += [
+            'user32.dll',
+        ]
+
     if CONFIG['MOZ_MEMORY'] and (CONFIG['MOZ_SYSTEM_JEMALLOC'] or FORCE_SHARED_LIB):
         pass
         # TODO: SHARED_LIBRARY_LIBS go here
-- 
1.9.1

