From aa5ddaf758997fe7ab1ddb457108866d2b3ac376 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Mon, 2 May 2016 15:37:00 +0200
Subject: Disable __thread on mingw


diff --git a/mozjs/js/src/old-configure.in b/mozjs/js/src/old-configure.in
index 7a0f359..0f0e587 100644
--- a/mozjs/js/src/old-configure.in
+++ b/mozjs/js/src/old-configure.in
@@ -1807,7 +1807,7 @@ if test "$ac_cv_thread_keyword" = yes; then
     mips*-*)
       :
       ;;
-    *-android*|*-linuxandroid*)
+    *-android*|*-linuxandroid*|*-mingw*)
       :
       ;;
     *)
-- 
1.9.1

