diff --git a/mozjs/config/external/nspr/moz.build b/mozjs/config/external/nspr/moz.build
index 18b9a1a5cf..231f6ec380 100644
--- a/mozjs/config/external/nspr/moz.build
+++ b/mozjs/config/external/nspr/moz.build
@@ -18,10 +18,11 @@ if CONFIG['MOZ_BUILD_NSPR']:
             'nss',
         ]
     else:
-        USE_LIBS += [
-            'nspr4',
-            'plc4',
-            'plds4',
-        ]
+        #USE_LIBS += [
+        #    'nspr4',
+        #    'plc4',
+        #    'plds4',
+        #]
+        pass
     EXPORTS.nspr += ['prcpucfg.h']
 else:
     OS_LIBS += CONFIG['NSPR_LIBS']