From 202297d41a2d1fc07a50ff952530c9117307874f Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 6 Apr 2016 10:38:42 +0200
Subject: Dumb down LinkedList for rust-bindgen


diff --git a/mozjs/mfbt/LinkedList.h b/mozjs/mfbt/LinkedList.h
index e0f2a03..a32305e 100644
--- a/mozjs/mfbt/LinkedList.h
+++ b/mozjs/mfbt/LinkedList.h
@@ -114,8 +114,8 @@ class LinkedListElement
    */
 
 private:
-  LinkedListElement* mNext;
-  LinkedListElement* mPrev;
+  LinkedListElement<T>* mNext;
+  LinkedListElement<T>* mPrev;
   const bool mIsSentinel;
 
 public:
-- 
1.9.1

