From 548db1584f4d358f651cbed419f3a020717184ec
Author: Vladimir Vukicevic <vladimir@pobox.com>
Date: Tue May 31 15:59:58 2016 -0400
Subject: Don't mix up struct/class in WinNSPR.cpp, as MSVC mangles them differently

diff --git a/mozjs/js/src/vm/WinNSPR.cpp b/mozjs/js/src/vm/WinNSPR.cpp
index b3ecd17..a67489d 100644
--- a/mozjs/js/src/vm/WinNSPR.cpp
+++ b/mozjs/js/src/vm/WinNSPR.cpp
@@ -70,8 +70,9 @@ SetThreadName(DWORD, const char*)
 }
 #endif
 
-struct nspr::Thread
+class nspr::Thread
 {
+public:
     HANDLE thread;
     DWORD threadId;
     void (*start)(void* arg);
