diff --git a/mozjs/config/system-headers.mozbuild b/mozjs/config/system-headers.mozbuild
index 0c5519849d..898827d4a2 100644
--- a/mozjs/config/system-headers.mozbuild
+++ b/mozjs/config/system-headers.mozbuild
@@ -1048,7 +1048,7 @@ system_headers = [
     'zmouse.h',
 ]
 
-if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'android':
+if CONFIG['OS_TARGET'] == 'Android':
     system_headers += [
         'android/api-level.h',
         'android/ashmem.h',
