From 37045452c3e88c062c101c2365d84e5a8a8a3697 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 6 Apr 2016 10:43:45 +0200
Subject: Hide unions in JSJitInfo

Need it in constant context.

diff --git a/mozjs/js/src/jsfriendapi.h b/mozjs/js/src/jsfriendapi.h
index 4d86c11..4ac8014 100644
--- a/mozjs/js/src/jsfriendapi.h
+++ b/mozjs/js/src/jsfriendapi.h
@@ -2337,6 +2337,9 @@ struct JSJitInfo {
         return JSValueType(returnType_);
     }
 
+#ifdef RUST_BINDGEN
+    const void *call;
+#else
     union {
         JSJitGetterOp getter;
         JSJitSetterOp setter;
@@ -2344,18 +2347,27 @@ struct JSJitInfo {
         /** A DOM static method, used for Promise wrappers */
         JSNative staticMethod;
     };
+#endif
 
+#ifdef RUST_BINDGEN
+    uint16_t protoID;
+#else
     union {
         uint16_t protoID;
         js::jit::InlinableNative inlinableNative;
     };
+#endif
 
+#ifdef RUST_BINDGEN
+    uint16_t depth;
+#else
     union {
         uint16_t depth;
 
         // Additional opcode for some InlinableNative functions.
         uint16_t nativeOp;
     };
+#endif
 
     // These fields are carefully packed to take up 4 bytes.  If you need more
     // bits for whatever reason, please see if you can steal bits from existing
-- 
1.9.1

