From af6efe5ed8bf5bbfa56729ba87a9b6c951a7d028 Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 6 Apr 2016 10:43:11 +0200
Subject: Hide some unions in JSPropertySpec

We need to use that type in constant context.

diff --git a/mozjs/js/src/jsapi.h b/mozjs/js/src/jsapi.h
index 0150093..8d13907 100644
--- a/mozjs/js/src/jsapi.h
+++ b/mozjs/js/src/jsapi.h
@@ -1989,6 +1989,10 @@ struct JSPropertySpec {
 
     const char*                 name;
     uint8_t                     flags;
+#ifdef RUST_BINDGEN
+    JSNativeWrapper             getter;
+    JSNativeWrapper             setter;
+#else
     union {
         JSNativeWrapper     native;
         SelfHostedWrapper   selfHosted;
@@ -1997,6 +2001,7 @@ struct JSPropertySpec {
         JSNativeWrapper           native;
         SelfHostedWrapper         selfHosted;
     } setter;
+#endif
 
     bool isSelfHosted() const {
 #ifdef DEBUG
-- 
1.9.1

