From 87ea8dbb227b86696d7a668630337e348e563faa Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 6 Apr 2016 10:42:36 +0200
Subject: Hide source hooks


diff --git a/mozjs/js/src/jsfriendapi.h b/mozjs/js/src/jsfriendapi.h
index 55a7953..4d86c11 100644
--- a/mozjs/js/src/jsfriendapi.h
+++ b/mozjs/js/src/jsfriendapi.h
@@ -448,6 +448,7 @@ class SourceHook {
     virtual bool load(JSContext* cx, const char* filename, char16_t** src, size_t* length) = 0;
 };
 
+#ifndef RUST_BINDGEN
 /**
  * Have |rt| use |hook| to retrieve lazily-retrieved source code. See the
  * comments for SourceHook. The runtime takes ownership of the hook, and
@@ -460,6 +461,7 @@ SetSourceHook(JSRuntime* rt, mozilla::UniquePtr<SourceHook> hook);
 /** Remove |rt|'s source hook, and return it. The caller now owns the hook. */
 extern JS_FRIEND_API(mozilla::UniquePtr<SourceHook>)
 ForgetSourceHook(JSRuntime* rt);
+#endif
 
 extern JS_FRIEND_API(JS::Zone*)
 GetCompartmentZone(JSCompartment* comp);
-- 
1.9.1

