From 1fa3b3e83d4159a08c566a3372839655280cae9f Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 6 Apr 2016 10:41:29 +0200
Subject: Hide TraceIncomingCCWs


diff --git a/mozjs/js/public/TracingAPI.h b/mozjs/js/public/TracingAPI.h
index 9472fb7..f8e3b4b 100644
--- a/mozjs/js/public/TracingAPI.h
+++ b/mozjs/js/public/TracingAPI.h
@@ -317,6 +317,7 @@ UnsafeTraceRoot(JSTracer* trc, T* edgep, const char* name);
 extern JS_PUBLIC_API(void)
 TraceChildren(JSTracer* trc, GCCellPtr thing);
 
+#ifndef RUST_BINDGEN
 using ZoneSet = js::HashSet<Zone*, js::DefaultHasher<Zone*>, js::SystemAllocPolicy>;
 using CompartmentSet = js::HashSet<JSCompartment*, js::DefaultHasher<JSCompartment*>,
                                    js::SystemAllocPolicy>;
@@ -328,6 +329,7 @@ using CompartmentSet = js::HashSet<JSCompartment*, js::DefaultHasher<JSCompartme
  */
 extern JS_PUBLIC_API(void)
 TraceIncomingCCWs(JSTracer* trc, const JS::CompartmentSet& compartments);
+#endif
 
 } // namespace JS
 
-- 
1.9.1

