From e69aedeef3a08ee059e7f36439794fbc3352a4cf Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 6 Apr 2016 10:41:54 +0200
Subject: Hide GarbageCollectionEvent


diff --git a/mozjs/js/public/GCAPI.h b/mozjs/js/public/GCAPI.h
index 5bedf24..e86d686 100644
--- a/mozjs/js/public/GCAPI.h
+++ b/mozjs/js/public/GCAPI.h
@@ -267,9 +267,13 @@ AbortIncrementalGC(JSRuntime* rt);
 
 namespace dbg {
 
-// The `JS::dbg::GarbageCollectionEvent` class is essentially a view of the
-// `js::gcstats::Statistics` data without the uber implementation-specific bits.
-// It should generally be palatable for web developers.
+/**
+ * The `JS::dbg::GarbageCollectionEvent` class is essentially a view of the
+ * `js::gcstats::Statistics` data without the uber implementation-specific bits.
+ * It should generally be palatable for web developers.
+ *
+ * <div rustbindgen hide></div>
+ */
 class GarbageCollectionEvent
 {
     // The major GC number of the GC cycle this data pertains to.
@@ -344,7 +348,9 @@ struct JS_PUBLIC_API(GCDescription) {
     char16_t* formatSummaryMessage(JSRuntime* rt) const;
     char16_t* formatJSON(JSRuntime* rt, uint64_t timestamp) const;
 
+#ifndef RUST_BINDGEN
     JS::dbg::GarbageCollectionEvent::Ptr toGCEvent(JSRuntime* rt) const;
+#endif
 };
 
 typedef void
-- 
1.9.1

