diff --git a/mozjs/js/src/jsapi.cpp b/mozjs/js/src/jsapi.cpp
index 94ef1b3ed..06be95062 100644
--- a/mozjs/js/src/jsapi.cpp
+++ b/mozjs/js/src/jsapi.cpp
@@ -5162,6 +5162,14 @@ JS::GetPromiseResult(JS::HandleObject promiseObj)
     return promise->state() == JS::PromiseState::Fulfilled ? promise->value() : promise->reason();
 }
 
+JS_PUBLIC_API(bool)
+JS::GetPromiseIsHandled(JS::HandleObject promiseObj)
+{
+    PromiseObject* promise = &promiseObj->as<PromiseObject>();
+    MOZ_ASSERT(promise->state() == JS::PromiseState::Rejected);
+    return !promise->isUnhandled();
+}
+
 JS_PUBLIC_API(JSObject*)
 JS::GetPromiseAllocationSite(JS::HandleObject promise)
 {
diff --git a/mozjs/js/src/jsapi.h b/mozjs/js/src/jsapi.h
index c7f92e273..5c5e9abc5 100644
--- a/mozjs/js/src/jsapi.h
+++ b/mozjs/js/src/jsapi.h
@@ -4390,6 +4390,12 @@ GetPromiseID(JS::HandleObject promise);
 extern JS_PUBLIC_API(JS::Value)
 GetPromiseResult(JS::HandleObject promise);
 
+/**
+ * Returns thhe given Promise's internal slot of `isHandled`.
+ */
+extern JS_PUBLIC_API(bool)
+GetPromiseIsHandled(JS::HandleObject promise);
+
 /**
  * Returns a js::SavedFrame linked list of the stack that lead to the given
  * Promise's allocation.
