From 2ba1bd3ff0ad3d30c5de67260ed2a8466a4b95da Mon Sep 17 00:00:00 2001
From: Michael Wu <mwu@mozilla.com>
Date: Thu, 23 Apr 2015 22:50:50 -0400
Subject: Linker hack


diff --git a/mozjs/js/src/old-configure.in b/mozjs/js/src/old-configure.in
index 6fede11..5788f7b 100644
--- a/mozjs/js/src/old-configure.in
+++ b/mozjs/js/src/old-configure.in
@@ -1897,8 +1897,7 @@ Android|WINNT|Darwin)
   ;;
 *)
   dnl On !Android !Windows !OSX, we only want to link executables against mozglue
-  MOZ_GLUE_IN_PROGRAM=1
-  AC_DEFINE(MOZ_GLUE_IN_PROGRAM)
+  MOZ_GLUE_IN_PROGRAM=
   ;;
 esac
 
