From 7bc2c1c7de47501dffc14ac370b5c3157e0e857e Mon Sep 17 00:00:00 2001
From: Anthony Ramine <n.oxyde@gmail.com>
Date: Wed, 27 Apr 2016 14:09:32 +0200
Subject: Make a horrible hack to avoid dllimport/dllexport on Windows


diff --git a/mozjs/mfbt/Types.h b/mozjs/mfbt/Types.h
index e7e18ab..5b6b59e 100644
--- a/mozjs/mfbt/Types.h
+++ b/mozjs/mfbt/Types.h
@@ -73,6 +73,10 @@
 #  define MOZ_IMPORT_DATA  MOZ_EXPORT
 #endif
 
+#if defined(_WIN32)
+#  define MFBT_API
+#  define MFBT_DATA
+#else
 /*
  * Consistent with the above comment, the MFBT_API and MFBT_DATA macros expose
  * export mfbt declarations when building mfbt, and they expose import mfbt
@@ -97,6 +101,7 @@
 #    define MFBT_DATA  MOZ_IMPORT_DATA
 #  endif
 #endif
+#endif
 
 /*
  * C symbols in C++ code must be declared immediately within |extern "C"|
diff --git a/mozjs/nsprpub/pr/include/prtypes.h b/mozjs/nsprpub/pr/include/prtypes.h
index 52b3ab0..3739631 100644
--- a/mozjs/nsprpub/pr/include/prtypes.h
+++ b/mozjs/nsprpub/pr/include/prtypes.h
@@ -142,6 +142,10 @@
 
 #endif
 
+#if defined(WIN32)
+#define NSPR_API(__type) __type
+#define NSPR_DATA_API(__type) __type
+#else
 #if defined(_NSPR_BUILD_)
 #define NSPR_API(__type) PR_EXPORT(__type)
 #define NSPR_DATA_API(__type) PR_EXPORT_DATA(__type)
@@ -149,6 +153,7 @@
 #define NSPR_API(__type) PR_IMPORT(__type)
 #define NSPR_DATA_API(__type) PR_IMPORT_DATA(__type)
 #endif
+#endif
 
 /***********************************************************************
 ** MACROS:      PR_BEGIN_MACRO
-- 
1.9.1

