From c162050717215f060e1f2e8dfe66a6a5ec7abd8f
Author: Vladimir Vukicevic <vladimir@pobox.com>
Date: Thu Jul 21 09:19:43 2016 -0400
Subject: Don't include pthread.h on Windows in PosixNSPR.h

diff --git a/mozjs/js/src/vm/PosixNSPR.h b/mozjs/js/src/vm/PosixNSPR.h
index edfd5fb..36a8e36 100644
--- a/mozjs/js/src/vm/PosixNSPR.h
+++ b/mozjs/js/src/vm/PosixNSPR.h
@@ -9,7 +9,9 @@
 
 #ifdef JS_POSIX_NSPR
 
+#ifndef _MSC_VER
 #include <pthread.h>
+#endif
 #include <stdint.h>
 
 namespace nspr {
