diff --git a/mozjs/js/src/vm/PosixNSPR.cpp b/mozjs/js/src/vm/PosixNSPR.cpp
index 41ce31e488..fdeee75b1b 100644
--- a/mozjs/js/src/vm/PosixNSPR.cpp
+++ b/mozjs/js/src/vm/PosixNSPR.cpp
@@ -11,7 +11,6 @@
 #ifdef JS_POSIX_NSPR
 
 #  include <errno.h>
-#  include <sys/time.h>
 #  include <time.h>
 
 int32_t PR_FileDesc2NativeHandle(PRFileDesc *fd) {
diff --git a/mozjs/js/src/vm/Time.cpp b/mozjs/js/src/vm/Time.cpp
index e1c810cc29..19ee15009b 100644
--- a/mozjs/js/src/vm/Time.cpp
+++ b/mozjs/js/src/vm/Time.cpp
@@ -27,8 +27,6 @@
 #  include <mmsystem.h> /* for timeBegin/EndPeriod */
 #  include <stdlib.h>   /* for _set_invalid_parameter_handler */
 
-#  include "prinit.h"
-
 #endif
 
 #ifdef XP_UNIX
