From bc064d6b54fd8a08dd3c759881167e8355d70aac
Author: Vladimir Vukicevic <vladimir@pobox.com>
Date: Tue May 31 16:00:35 2016 -0400
Subject: When doing RUST_BINDGEN, we'll be using clang/llvm, and should use those visibillity attributes

diff --git a/mozjs/mfbt/Types.h b/mozjs/mfbt/Types.h
index 5b6b59e..abba6cf 100644
--- a/mozjs/mfbt/Types.h
+++ b/mozjs/mfbt/Types.h
@@ -38,7 +38,11 @@
  * methods or data used cross-file.
  */
 #if defined(WIN32)
-#  define MOZ_EXPORT   __declspec(dllexport)
+#  ifdef RUST_BINDGEN
+#    define MOZ_EXPORT   __attribute__((visibility("default")))
+#  else
+#    define MOZ_EXPORT   __declspec(dllexport)
+#  endif
 #else /* Unix */
 #  ifdef HAVE_VISIBILITY_ATTRIBUTE
 #    define MOZ_EXPORT       __attribute__((visibility("default")))
