commit 9d3e010f0e757dc2e4294da39a7548bbeb049366
Author: Josh Matthews <josh@joshmatthews.net>
Date:   Thu Jan 2 10:50:27 2020 -0500

    Support profiling feature when building for UWP.

diff --git a/mozjs/js/moz.configure b/mozjs/js/moz.configure
index 5d99f7155a..e87080bf38 100644
--- a/mozjs/js/moz.configure
+++ b/mozjs/js/moz.configure
@@ -271,13 +271,13 @@ def profiling(value):
 
 add_old_configure_assignment('MOZ_PROFILING', profiling)
 
-@depends(profiling, target)
-def imply_vtune(value, target):
+@depends(profiling, target, js_enable_uwp)
+def imply_vtune(value, target, uwp):
     ok_cpu    = target.cpu in ['x86', 'x86_64']
     ok_kernel = target.kernel == 'WINNT' or \
                 (target.kernel == 'Linux' and target.os == 'GNU')
 
-    if value and ok_cpu and ok_kernel:
+    if value and ok_cpu and ok_kernel and not uwp:
         return True
 
 set_config('MOZ_PROFILING', profiling)
diff --git a/mozjs/js/src/builtin/Profilers.cpp b/mozjs/js/src/builtin/Profilers.cpp
index 53da01dba9..6b2f848510 100644
--- a/mozjs/js/src/builtin/Profilers.cpp
+++ b/mozjs/js/src/builtin/Profilers.cpp
@@ -24,8 +24,13 @@
 #endif
 
 #ifdef XP_WIN
-#  include <process.h>
-#  define getpid _getpid
+#  ifdef JS_ENABLE_UWP
+#    include <processthreadsapi.h>
+#    define getpid GetCurrentProcessId
+#  else
+#    include <process.h>
+#    define getpid _getpid
+#  endif
 #endif
 
 #include "js/CharacterEncoding.h"
