From 2e8cfd9bc9ea91dd0bd6ff09c0963fb5dda12819 Mon Sep 17 00:00:00 2001
From: Vladimir Vukicevic <vladimir@pobox.com>
Date: Thu, 24 Sep 2015 14:06:00 -0400
Subject: win32: allow msys make when in mingw64/msys2


diff --git a/mozjs/config/baseconfig.mk b/mozjs/config/baseconfig.mk
index a125466..368c268 100644
--- a/mozjs/config/baseconfig.mk
+++ b/mozjs/config/baseconfig.mk
@@ -25,9 +25,12 @@ ifdef .PYMAKE
 $(error Pymake is no longer supported. Please upgrade to MozillaBuild 1.9 or newer and build with 'mach' or 'mozmake')
 endif
 
+# don't support MSYS make, except if we're actually under mingw64/msys2
+ifneq ($(MSYSTEM),MINGW64)
 ifeq (a,$(firstword a$(subst /, ,$(abspath .))))
 $(error MSYS make is not supported)
 endif
+endif
 # 4.0- happens to be greater than 4.0, lower than the mozmake version,
 # and lower than 4.0.1 or 4.1, whatever next version of gnu make will
 # be released.
-- 
1.9.1

