diff --git a/js/moz.configure b/js/moz.configure
index e65adafb73..d9590f1f8b 100644
--- a/js/moz.configure
+++ b/js/moz.configure
@@ -54,7 +54,6 @@ set_config("JS_ENABLE_SMOOSH", enable_smoosh)
 set_define("JS_ENABLE_SMOOSH", enable_smoosh)
 
 include("../build/moz.configure/nspr.configure", when="--enable-compile-environment")
-include("../build/moz.configure/rust.configure", when="--enable-compile-environment")
 include("../build/moz.configure/bindgen.configure", when="--enable-compile-environment")
 
 set_config("JS_STANDALONE", js_standalone)
