diff --git a/js/moz.configure b/js/moz.configure
index e65adafb73..48ec64d925 100644
--- a/js/moz.configure
+++ b/js/moz.configure
@@ -290,14 +290,14 @@ with only_when("--enable-compile-environment"):
     imply_option("--enable-frame-pointers", True, when=profiling)
 
 
-@depends(profiling, target)
-def imply_vtune(value, target):
+@depends(profiling, target, js_enable_uwp)
+def imply_vtune(value, target, uwp):
     ok_cpu = target.cpu in ["x86", "x86_64"]
     ok_kernel = target.kernel == "WINNT" or (
         target.kernel == "Linux" and target.os == "GNU"
     )
 
-    if value and ok_cpu and ok_kernel:
+    if value and ok_cpu and ok_kernel and not uwp:
         return True
 
 
