commit 9c67745089c18bc24823435d71c40bfed575ea62
Author: Josh Matthews <josh@joshmatthews.net>
Date:   Sun Jan 31 22:13:20 2021 -0500

    force python to load virtualenv module from subdir

diff --git a/third_party/python/virtualenv/virtualenv.py b/third_party/python/virtualenv/virtualenv.py
index f6c8ce7d1a..9b92127307 100755
--- a/third_party/python/virtualenv/virtualenv.py
+++ b/third_party/python/virtualenv/virtualenv.py
@@ -46,6 +46,7 @@ def run():
     with mod.VersionedFindLoad() as finder:
         sys.meta_path.insert(0, finder)
         finder._register_distutils_finder()
+        sys.path.insert(0, os.path.join(os.path.dirname(__file__)))
         from virtualenv.__main__ import run as run_virtualenv
 
         run_virtualenv()
