diff --git a/mozglue/misc/moz.build b/mozglue/misc/moz.build
index 1b79cbaf6a..4bb8f5a110 100644
--- a/mozglue/misc/moz.build
+++ b/mozglue/misc/moz.build
@@ -33,7 +33,6 @@ EXPORTS.mozilla.glue += [
 if CONFIG["OS_ARCH"] == "WINNT":
     EXPORTS.mozilla += [
         "GetKnownFolderPath.h",
-        "PreXULSkeletonUI.h",
         "StackWalk_windows.h",
         "StackWalkThread.h",
         "TimeStamp_windows.h",
@@ -105,7 +104,6 @@ if CONFIG["OS_ARCH"] == "WINNT":
         SOURCES += [
             "/ipc/mscom/COMWrappers.cpp",
             "/ipc/mscom/ProcessRuntime.cpp",
-            "PreXULSkeletonUI.cpp",
         ]
 
     if CONFIG["ACCESSIBILITY"]:
