diff --git a/memory/mozalloc/moz.build b/memory/mozalloc/moz.build
index 2d13c6cf14..5ce2ccf751 100644
--- a/memory/mozalloc/moz.build
+++ b/memory/mozalloc/moz.build
@@ -21,12 +21,6 @@ if CONFIG["WRAP_STL_INCLUDES"]:
             "msvc_raise_wrappers.cpp",
         ]
 
-if CONFIG["OS_TARGET"] == "WINNT":
-    # Keep this file separate to avoid #include'ing windows.h everywhere.
-    SOURCES += [
-        "winheap.cpp",
-    ]
-
 UNIFIED_SOURCES += [
     "mozalloc.cpp",
     "mozalloc_abort.cpp",
