diff --git a/mozjs/mozjs/js/src/util/GetPidProvider.h b/mozjs/mozjs/js/src/util/GetPidProvider.h
index ca00c5bca1..13cd39ead9 100644
--- a/mozjs/mozjs/js/src/util/GetPidProvider.h
+++ b/mozjs/mozjs/js/src/util/GetPidProvider.h
@@ -8,8 +8,13 @@
 #define util_GetPidProvider_h
 
 #ifdef XP_WIN
-#  include <process.h>
-#  define getpid _getpid
+#  ifdef JS_ENABLE_UWP
+#    include <processthreadsapi.h>
+#    define getpid GetCurrentProcessId
+#  else
+#    include <process.h>
+#    define getpid _getpid
+#  endif
 #else
 #  include <unistd.h>
 #endif
